python/uyuni/common/rhn_rpm.py

```diff
@@ -151,11 +151,13 @@ def checksum_type(self):
         return checksum_type
 
     def is_signed(self):
+        dsaheader = rsaheader = 0
         if hasattr(rpm, "RPMTAG_DSAHEADER"):
             dsaheader = self.hdr["dsaheader"]
-        else:
-            dsaheader = 0
-        if self.hdr["siggpg"] or self.hdr["sigpgp"] or dsaheader:
+        if hasattr(rpm, "RPMTAG_RSAHEADER"):
+            rsaheader = self.hdr["rsaheader"]
+
+        if self.hdr["siggpg"] or self.hdr["sigpgp"] or dsaheader or rsaheader:
```

Comment:
I haven't checked too deep, but not sure if `siggpg` or `sigpgp` are always present in `self.hdr`, maybe this way is safer?
```suggestion
        if self.hdr.get("siggpg") or self.hdr.get("sigpgp") or dsaheader or rsaheader:
```

Reply:
self.hdr is not just a dict(). I have no idea what object it is. So I would like to keep it here. This statement works since 2008. I think we are good for now.